### ISSUE TITLE
[Bug]: Unexpected error in console due to current `isFunctionAsync` check

### ISSUE THREAD

**Issue by Rishabh-Rathod** [appsmithorg/appsmith#12639]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

<img width="1394" alt="Screenshot 2022-04-06 at 11 14 55 PM" src="https://user-images.githubusercontent.com/23132741/162036306-9748ce1b-d065-4577-bf28-13d8c1d9704f.png">

### Steps To Reproduce

1. Create an app with a Filepicker widget and an API name `Api1`
2. Create a JS Object with the code mentioned below
```
export default {
	myFun2:() => {
		console.log(FilePicker1.files);
		Api1.run().then(()=>{
      showAlert("hello");
    })
	}
}
```
3. Refresh the page and notice the dev tools console log

### Public Sample App

https://release.app.appsmith.com/console-errors-isasync/page1-624daa8fdbe66e13e8d84ad4

### Version

1.6.17

**Comment by Rishabh-Rathod** [comment]
It was initially noticed in users' applications where we noticed more than 10 such errors in the console. It was confusing for users to determine if the application actually has error or not. 
Also, console.log messages of unEval values were not expected and were making debugging hard.

**Comment by ramsaptami** [comment]
Calling a query inside a JS object that's been marked as async still shows the `Async in sync` field error on console.

Steps to repro: Type in following code into a JS object, refresh page and observe console for the error
```
export default {
	myFun1: async () => {
		SelectQuery1.run()
	}
}
```

**Comment by Rishabh-Rathod** [comment]
@rohan-arthur as this issue is moved from product backlog to in analysis, do we want to add RACI metric to this issue? 
- All the users who write JS code and use browser dev tools for checking logs will be what we could consider in reach.

**Comment by rohan-arthur** [comment]
@Rishabh-Rathod can this be fixed along with #12179 ? if yes, we can include the issue in the epic
